File: test_drive/lib/widgets/hybrid_editor/wikilinks/wiki_link_trigger.dart
class WikiTrigger {
  static int? findStart(String text, int caret) {
    if (caret < 2) return null;
    final index = text.lastIndexOf('[[', caret - 1);
    return index;
  }

  static bool shouldStart(String text, int caret) {
    return findStart(text, caret) != null;
  }
}

File: native/components/editor/wikilinks/WikiLinkTrigger.ts
/// Utility functions for detecting wiki link triggers ([[)
export class WikiLinkTrigger {
  /// Finds the start position of a wiki link trigger before the caret
  /// Returns null if no trigger is found
  static findStart(text: string, caret: number): number | null {
    if (caret < 2) return null;
    const index = text.lastIndexOf('[[', caret - 1);
    return index >= 0 ? index : null;
  }
}

